Application/EdFi.Ods.Standard/Security/Authorization/Overrides/EducationOrganizationNetworkAssociationRelationshipsAuthorizationContextDataProvider.cs

```diff
@@ -0,0 +1,59 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using EdFi.Ods.Api.Security.AuthorizationStrategies.Relationships;
+using EdFi.Ods.Entities.Common.EdFi;
+using EdFi.Ods.Entities.NHibernate.EducationOrganizationNetworkAssociationAggregate.EdFi;
+
+namespace EdFi.Ods.Standard.Security.Authorization.Overrides
+{
+    /// <summary>
+    /// Creates and returns an <see cref="RelationshipsAuthorizationContextData"/> instance for making authorization decisions for access to the edfi.EducationOrganizationNetworkAssociation table of the EducationOrganizationNetworkAssociation aggregate in the Ods Database.
+    /// </summary>
+    [ExcludeFromCodeCoverage]
+    public class EducationOrganizationNetworkAssociationRelationshipsAuthorizationContextDataProvider<TContextData> : IRelationshipsAuthorizationContextDataProvider<IEducationOrganizationNetworkAssociation, TContextData>
+        where TContextData : RelationshipsAuthorizationContextData, new()
+    {
+        /// <summary>
+        /// Creates and returns an <see cref="TContextData"/> instance based on the supplied resource.
+        /// </summary>
+        public TContextData GetContextData(IEducationOrganizationNetworkAssociation resource)
+        {
+            if (resource == null)
+                throw new ArgumentNullException("resource", "The 'educationOrganizationNetworkAssociation' resource for obtaining authorization context data cannot be null.");
+
+            var entity = resource as EducationOrganizationNetworkAssociation;
+
+            var contextData = new TContextData();
+            // contextData.EducationOrganizationNetworkId = entity.EducationOrganizationNetworkId == default(long) ? null as long? : entity.EducationOrganizationNetworkId; // Primary key property, Only Education Organization Id present
+            contextData.EducationOrganizationId = entity.MemberEducationOrganizationId; // Primary key property, Role name applied
```

Comment:
I added this override because smoke tests failed on the EducationOrganizationNetworkAssociation Post, with an Authorization Denied error indicating that the caller's EducationOrganization had no relationship to the EducationOrganizationNetworkId; so I looked at the database table for the Association and added this override based on the MemberEducationOrganization foreign key.

With this, I'm not sure if there is an error in the smoke test code and the EducationOrganizationNetworkAssociation was included as a valid EdOrg or if it still can be solved with an override.

Reply:
We definitely need to revert this approach and find a different solution that doesn't involve changing the authorization context data used for authorizing the resource.